test/features/bedEvent.js

```diff
@@ -0,0 +1,713 @@
+'use strict';
+
+var BedEvent2 = require('../../api/models/BedEvent');
+const centre = {
+  name: 'testexample',
+  male_capacity: 999,
+  female_capacity: 750,
+  id: 1,
+  male_cid_name: ['testexample male holding', 'smallone male holding'],
+  female_cid_name: ['bigone female office', 'smallone female holding']
+};
+const bedRef = 'testExample';
+var timestamp = {
+  newest: "2017-11-23T18:25:43.511Z",
+  new: "2016-11-23T18:25:42.511Z",
+  old: "2015-11-23T18:25:41.511Z",
+  oldest: "2014-11-23T18:25:40.511Z"
+};
+
+var findBedEvent = (event, active) =>
+  BedEvent
+    .find({
+      where: {
+        timestamp: event.timestamp,
+        operation: event.operation,
+        active: active
+      }
+    })
+    .populate('bed', {where: {bed_ref: event.bed_ref}})
+    .toPromise()
+    .filter((bedEvents) => !_.isEmpty(bedEvents.bed));
+
+Feature('Bed Events', () => {
+  Scenario('Old OOC then New IC', () => {
+    var payload = {
+      OUT_OF_COMMISSION: {
+        operation: BedEvent2.OPERATION_OUT_OF_COMMISSION,
+        timestamp: timestamp.old,
+        centre: centre.name,
+        bed_ref: bedRef,
+        reason: 'Other',
+        gender: 'm'
+      },
+      IN_COMMISSION: {
+        operation: BedEvent2.OPERATION_IN_COMMISSION,
+        timestamp: timestamp.new,
+        centre: centre.name,
+        bed_ref: bedRef
+      }
+    };
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    Given('there is a single centre', () =>
+      Centres.create(centre)
+        .then(() => expect(Centres.find({id: centre.id})).to.eventually.have.lengthOf(1))
+    );
+
+    And(`an OOC Bed Event with bed ref "${bedRef}" & timestamp "${timestamp.old}" occurs`,
+      () => createRequest(payload.OUT_OF_COMMISSION, '/irc_entry/event', 201)
+    );
+
+    When(`a new IC Bed Event with bed ref "${bedRef}" & timestamp "${timestamp.new}" occurs`,
+      () => createRequest(payload.IN_COMMISSION, '/irc_entry/event', 201)
+    );
+
+    Then(`IC Bed Event with bed ref "${bedRef}" is marked as active`,
+      () => findBedEvent(payload.IN_COMMISSION, true)
```

Comment:
please be consistent with where the new line goes, we appear to elsewhere put the `() =>` on the same line as the `Given/When/Then/And`